### PR TITLE
Treat warnings as errors

### PR DESCRIPTION
**Pull request by twollgam**
### What does this PR do?

To avoid introducing new warnings they will be treated as errors. 

### Closes Issue(s)

None

### Motivation

Make it easier for developers to find warnings.

### CODE REVIEW

File: game/CMakeLists.txt [hunk 1]
```diff
@@ -311,6 +311,12 @@ if(WIN32)
 	set_target_properties(performous PROPERTIES OUTPUT_NAME "Performous")
 endif()
 
+if(WIN32 AND MSVC)
+	target_compile_options(performous PUBLIC /WX)
+else()
+	target_compile_options(performous PUBLIC -Werror)
```
Comment (Baklap4) [inline game/CMakeLists.txt:317]: just wondering if this also works for clang?
If yes let's go!

Comment (Baklap4) [reply game/CMakeLists.txt:317]: Might be better to just heck for MSVC isntead of WIN32, as MSVC _can_ be used on unix....

Comment (twollgam) [reply game/CMakeLists.txt:317]: > just wondering if this also works for clang? If yes let's go!

It should cause clang uses gccs command line AFAIK

Comment (twollgam) [reply game/CMakeLists.txt:317]: > Might be better to just heck for MSVC isntead of WIN32, as MSVC _can_ be used on unix right?

Not that I know. msvc is visual studio only. There is a mac version, but it only supports .net and no c++ AFAIK.

Comment (Baklap4) [reply game/CMakeLists.txt:317]: 👍

**Review by Baklap4 — APPROVED** [review]

**Comment by Baklap4** [comment]
If we wanna compile with treat warnings as errors on windows we need to specify a warning level instead of `/Wall` i think `/W4` is more than plenty for now @twollgam

**Pull request merged**